src/main/java/de/tum/in/www1/exerciseapp/web/rest/ProgrammingExerciseResource.java

```diff
@@ -79,19 +113,35 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
      * @return the ResponseEntity with status 200 (OK) and the list of programmingExercises in body
      */
     @GetMapping("/programming-exercises")
+    @PreAuthorize("hasAnyRole('TA', 'ADMIN')")
     @Timed
     public List<ProgrammingExercise> getAllProgrammingExercises() {
         log.debug("REST request to get all ProgrammingExercises");
         return programmingExerciseRepository.findAll();
         }
 
     /**
+     * GET  /courses/:courseId/exercises : get all the exercises.
+     *
+     * @return the ResponseEntity with status 200 (OK) and the list of programmingExercises in body
+     */
+    @GetMapping(value = "/courses/{courseId}/programming-exercises")
+    @PreAuthorize("hasAnyRole('TA', 'ADMIN')")
+    @Timed
+    @Transactional(readOnly = true)
+    public List<ProgrammingExercise> getProgrammingExercisesForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get a page of Exercises");
```

Comment:
The log statement is quite generic --> would be better to be specific with "Programming Exercises"